app/src/main/java/com/github/bjolidon/bootcamp/database/UserCardsRTDB.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.bjolidon.bootcamp.database
+
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.github.bjolidon.bootcamp.R
+import com.github.bjolidon.bootcamp.model.MagicCard
+import com.github.bjolidon.bootcamp.model.MagicLayout
+import com.github.bjolidon.bootcamp.model.MagicSet
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.CompletableFuture
+
+/**
+ * This class is used to manage the user's collection of cards.
+ */
+
+class UserCardsRTDB : AppCompatActivity(){
+    private val db = Firebase.database.reference
+    private val user = Firebase.auth.currentUser
+    private val userCardCollection = db.child(user!!.uid)
+
+    /**
+     * Adds a card to the user's collection.
+     */
+    fun addCardToCollection(card: MagicCard) : String {
+        val cardUID = card.set.code + card.number
+        userCardCollection.child(cardUID).child("name").setValue(card.name)

```

Comment:
can there be errors while setting those values ?

Reply:
The code at the moment assumes the card passed to the function is correct. setValue either adds the value and creates a new child if it doesnt exist already under that key, or it overwrites the old value if the key already exists (thats why its important to have unique IDs for each card)

Reply:
i understand, but setting this value can not fail ? this is nice !

Reply:
i just checked, the setValue(...) can return a task which can fail or succeed. Will be needed to handle cancellation and failure maybe

Reply:
then put a todo in comment to remember to do it. It s gonna be a task for next sprint